fcrepo-auth-roles-common/src/main/java/org/fcrepo/auth/roles/common/AbstractRolesAuthorizationDelegate.java

```diff
@@ -62,18 +63,13 @@
      * @param principals effective principals
      * @return set of effective content roles
      */
-    public static Set<String>
-    resolveUserRoles(final Map<String, List<String>> acl,
-                    final Set<Principal> principals) {
-        final Set<String> roles = new HashSet<>();
-        for (final Principal p : principals) {
-            final List<String> matchedRoles = acl.get(p.getName());
-            if (matchedRoles != null) {
-                LOGGER.debug("request principal matched role assignment: {}", p.getName());
-                roles.addAll(matchedRoles);
-            }
-        }
-        return roles;
+    public static Set<String> resolveUserRoles(final Map<String, Collection<String>> acl,
+                    final Collection<Principal> principals) {
+        return principals.stream().map(Principal::getName).filter(acl::containsKey)
```

Comment:
This is a pretty direct translation of the earlier function: in the earlier version, `acl.get(p.getName())` returns `matchedRoles`, which is null if the principal name doesn't exist in the map. The code then had an if block to exclude any null values. In the streamed version, we just filter out any principal value that doesn't exist in the acl Map before retrieving the value.